engine/src/main/java/com/datatorrent/stram/cli/ApexCli.java

```diff
@@ -2193,9 +2202,11 @@ public void execute(String[] args, ConsoleReader reader) throws Exception
       } else {
         apps = new ApplicationReport[args.length - 1];
         for (int i = 1; i < args.length; i++) {
-          apps[i - 1] = getApplication(args[i]);
-          if (apps[i - 1] == null) {
-            throw new CliException("Streaming application with id " + args[i] + " is not found.");
+          ApplicationReport ap = findApplicationReportFromAppNameOrId(args[i]);
+          if (ap == null) {
+            throw new CliException("Streaming application with id or name " + args[i] + " is not found.");
```

Comment:
Pls see my comment above about failing the command for the whole list when a single app is not found.